flow/connectors/postgres/qrep.go

```diff
@@ -68,13 +65,21 @@ func (c *PostgresConnector) GetQRepPartitions(
 	if err != nil {
 		return nil, fmt.Errorf("failed to begin transaction: %w", err)
 	}
+	// rollback transaction will no-op if transaction is committed
 	defer shared.RollbackTx(getPartitionsTx, c.logger)
 
 	if err := c.setTransactionSnapshot(ctx, getPartitionsTx, config.SnapshotName); err != nil {
 		return nil, fmt.Errorf("failed to set transaction snapshot: %w", err)
 	}
 
-	return c.getNumRowsPartitions(ctx, getPartitionsTx, config, last)
+	partitions, err := c.getPartitions(ctx, getPartitionsTx, config, last)
+
+	// commit transaction
+	if err := getPartitionsTx.Commit(ctx); err != nil {
+		return nil, fmt.Errorf("failed to commit transaction: %w", err)
+	}
```

Comment:
also open to just remove this altogether -- since this transaction is used for read-only purpose -- having `defer shared.RollbackTx()` above is sufficient.